### DISCUSSION TITLE
Content not dynamically resizing

### DISCUSSION THREAD

**Discussion by jaeaton** [AvaloniaUI/Avalonia#10666] [Q&A] [answered]
I have a control that I want to stretch with available height (IE when the window resizes I want the content of that control, a graph, to get smaller or bigger). This worked when it was a single control. I have made modifications to allow for multiple controls to be inserted as needed using data templates in the XAML.
<DataTemplate DataType="model:WinchModel">
   <Grid>
      <view:DataDisplay />
   </Grid>
</DataTemplate>

  I have then used and <ItemsControl  Items={ Binding WinchList }/> to populate the panel. 

The problem is that the content is the minimum height and not the available height. The unconstrained control is minimized to zero. If I set a minimum height it is the minimum height. If more items than can fit in the view are to be shown, they run off the view. If the window is resized the content stays at a fixed size.

I am guessing this has to do with the use of ItemsControl. Is there another container that can be used?

**Comment by timunie** [comment]
I can' understand the issue tbh. Why do you need an ItemsControl if you want to display only one item at a time? You could just add a COntentControl and let DataTemplates do the visual representation:

```xml
<ContentControl Content="{Binding CurrentItem}">
   <ContentControl.DataTemplates> 
     <!-- DataTemplates go here -->
   </ContentControl.DataTemplates>
</ContentControl>
```

If you really need to use ItemControl with Items having the height of the outer container, you can do this: 

```xml
<DataTemplate DataType="myDataType">
    <view:DataDisply Height="{Binding $parent[ItemsControl].Bounds.Height}" /> 
</DataTemplate>
```

Note: Bounds.Height is the actual height of the control in it's bound, so it should respect the available size

**Reply by jaeaton** [reply]
It looks like I was a bit unclear. I have multiple data sources that I want to be able to display. Not all need to be displayed at the same time. The number of plots can be chosen by the user (IE using a single instrument have a single plot, using multiple instruments have multiple plots). 

I left off my data template from my previous message. 

`<UserControl.DataTemplates>
		<DataTemplate DataType="model:WinchModel">
			<Grid>
				<controls:DataDisplayView />				
			</Grid>
		</DataTemplate>
	</UserControl.DataTemplates>`

Using `<view:DataDisplay Height="{Binding $parent[ItemsControl].Bounds.Height}" /> ` Results in plots being on top of each other with only one being visible at a time.

**Reply by timunie** [reply]
Probably you are looking for a Docking Layout? Check out: 
 - https://github.com/wieslawsoltes/Dock 
 - https://github.com/AvaloniaCommunity/awesome-avalonia#controls

**Comment by jaeaton** [comment]
The window is sized to the content
![image](https://user-images.githubusercontent.com/47159608/225354310-9629dbe2-7f0a-4818-ad8f-a3e229e50cc0.png)

Making the window larger the content doesn't scale
![image](https://user-images.githubusercontent.com/47159608/225354542-23a316fb-b264-4b08-bb6c-fb210193dfa9.png)

Multiple plots extend beyond the window
![image](https://user-images.githubusercontent.com/47159608/225355251-ac546216-aba1-42a5-b36e-1efbf1db7825.png)

**Reply by timunie** [reply]
I cannot help you further without debugging on my own. If you have the program open source, I probably can take a look. Otherwise you need to provide a minimal sample or figure out on your own. 

DevTools are helpful. Open them via F12

**Reply by jaeaton** [reply]
I appreciate your help. It is likely that I am missing something or have misunderstood something.

https://github.com/jaeaton/ECWP-Winch-Data/tree/5_0_0_b2

**Comment by timunie** [comment, marked as answer]
Update from the samples app: 

> Yeah honestly I still don't quite understand the desired output, but I assume you want several items below each other without a ScrollViewer at all. One Option I see is to use a  `UniformGrid` as ItemsControl.ItemPanel and remove the ScrollViewer completely. But this will require that you bind `UnifromGrid.Rows` to `Items.Count`. Also when you have too many items information may be missing. Anyway, Here is some sample code. 

```xml
<ItemsControl Items="{Binding TestItems}" >
    <ItemsControl.ItemTemplate>
        <DataTemplate>
           [Up to you]
        </DataTemplate>
    </ItemsControl.ItemTemplate>
    <ItemsControl.ItemsPanel>
        <ItemsPanelTemplate>
            <UniformGrid Rows="{Binding TestItems.Count}"></UniformGrid>
        </ItemsPanelTemplate>
    </ItemsControl.ItemsPanel>
</ItemsControl>
```

**Comment by jaeaton** [comment]
Tim,

Appreciate your help on this. This performs as desired.

Josh
